src/components/Chart/Table.js

```diff
@@ -0,0 +1,255 @@
+    marginLeft: '-20px',
+    marginRight: '-20px'
+  }),
+  table: css({
+    ...sansSerifRegular18,
+    lineHeight: '1.2',
+    minWidth: '100%',
+    borderSpacing: '0 2px',
+    borderCollapse: 'separate',
+    padding: '0 10px'
+  }),
+  header: css({
+    borderBottomWidth: '1px',
+    borderBottomStyle: 'solid',
+    padding: '8px 0',
+    userSelect: 'none',
+    '&:first-of-type': {
+      padding: '8px 0 8px 10px'
+    },
+    '&:last-of-type': {
+      padding: '8px 10px 8px 0'
+    }
+  }),
+  cell: css({
+    padding: '8px 0',
+    '&:first-of-type': {
+      padding: '8px 0 8px 10px'
+    },
+    '&:last-of-type': {
+      padding: '8px 10px 8px 0'
+    }
+  })
+}
+
+const Table = props => {
+  const [colorScheme] = useColorContext()
+  const {
+    values,
+    numberFormat,
+    colorRanges,
+    colorRange,
+    defaultSortColumn,
+    thresholds,
+    tableColumns
+  } = props
+  const columns = values.columns
+
+  const [sortBy, setSortBy] = useState({
+    key: defaultSortColumn,
+    order: 'desc',
+    enableSorting: !!defaultSortColumn || false
+  })
+
+  let parsedData = []
+
+  values.forEach(row => {
+    let parsedItem = {}
+    Object.keys(row).forEach(
+      item =>
+        (parsedItem[item] =
+          tableColumns.find(d => d.column === item)?.type === 'number'
+            ? +row[item]
+            : row[item])
+    )
+    parsedData.push({ ...parsedItem })
+  })
+
+  const sortedData = parsedData.sort((a, b) => {
+    if (!sortBy.enableSorting) {
+      return parsedData
+    }
+    if (typeof a[sortBy.key] === 'string') {
+      return sortBy.order === 'desc'
+        ? a[sortBy.key].localeCompare(b[sortBy.key])
+        : b[sortBy.key].localeCompare(a[sortBy.key])
+    }
+    return sortBy.order === 'desc'
+      ? b[sortBy.key] - a[sortBy.key]
+      : a[sortBy.key] - b[sortBy.key]
+  })
+
+  // helper function that toggles order (desc/asc) or sets new sort by key (order: desc)
+  const setSort = key => {
+    if (sortBy.key === key) {
+      setSortBy({
+        key,
+        order: sortBy.order === 'desc' ? 'asc' : 'desc',
+        enableSorting: true
+      })
+    } else {
+      setSortBy({ key, order: 'desc', enableSorting: true })
+    }
+  }
+
+  let currentColorRange = colorRanges[colorRange] || colorRange
+
+  const colorScale = (type, column) => {
+    let scale
+    let domain
+    if (type === 'number') {
+      if (thresholds) {
+        scale = scaleThreshold()
+        domain = thresholds
+        if (!colorRange) {
+          currentColorRange = colorRanges.sequential.slice(0, domain.length + 1)
+        }
+      } else {
+        scale = scaleQuantize()
+        domain = extent(parsedData, d => d[column])
+      }
+    } else {
+      scale = scaleOrdinal()
+      domain = parsedData.map(d => d[column])
+      currentColorRange = colorRanges.discrete.slice(0, domain.length + 1)
+    }
+    return scale
+      .domain(domain)
+      .range(currentColorRange || colorRanges.sequential)
+  }
```

Comment:
this is a bit crazy: we create a scale per column? this could mean we'd have two different color scales in action in the same table? one for number columns and an ordinal for string columns? And only the `number` columns support a colorRange defined outside? And no support for `colorMap` at all?

IMO just create one color scale. If `thresholds` is set use `scaleThreshold`, if not `scaleOrdinal`. If you want `scaleQuantize` support first add an example and then maybe set it behind the following condition: `tableColumns.filter(c => c.color).every(c => c.type === 'number')`

Reply:
what would be the domain then for scaleOrdinal?

Reply:
Something like this:
```
const colorColumns = tableColumns.filter(c => c.color).map(c => c.column)
const colorValues = !colorColumns.length ? [] : parsedData.reduce(
  (values, row) => {
    colorColumns.forEach(key => {
      values.push(row[key])
    })
    return values
  }
  []
).filter(Boolean).filter(deduplicate)

const color = getColorMapper(props, colorValues)
```